pilot/pkg/networking/v1alpha3/authentication.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha3
+
+import (
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	"github.com/golang/protobuf/ptypes/duration"
+
+	authn "istio.io/api/authentication/v1alpha1"
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/proxy/envoy/v1"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	// jwtFilterName is the name for the Jwt filter. This should be the same
+	// as the name defined in
+	// https://github.com/istio/proxy/blob/master/src/envoy/http/jwt_auth/http_filter_factory.cc#L50
+	jwtFilterName = "jwt-auth"
+)
+
+// buildJwtFilter returns a Jwt filter for all Jwt specs in the policy.
+func buildJwtFilter(policy *authn.Policy) *http_conn.HttpFilter {
+	filterConfigProto := model.ConvertPolicyToJwtConfig(policy)
+	if filterConfigProto == nil {
+		return nil
+	}
+	// TODO(diemtvu): check what is the right way to set config using proto.
+	return buildHTTPFilterConfig(jwtFilterName, filterConfigProto.String())
+}
+
+// buildJwksURIClustersForProxyInstances checks the authentication policy for the
+// input proxyInstances, and generates (outbound) clusters for all JwksURIs.
+func buildJwksURIClustersForProxyInstances(mesh *meshconfig.MeshConfig,
+	store model.IstioConfigStore, proxyInstances []*model.ServiceInstance) v1.Clusters {
+	if len(proxyInstances) == 0 {
+		return nil
+	}
+	var jwtSpecs []*authn.Jwt
+	for _, instance := range proxyInstances {
+		authnPolicy := model.GetConsolidateAuthenticationPolicy(mesh, store, instance.Service.Hostname, instance.Endpoint.ServicePort)
+		jwtSpecs = append(jwtSpecs, model.CollectJwtSpecs(authnPolicy)...)
+	}
+
+	return buildJwksURIClusters(jwtSpecs, mesh.ConnectTimeout)
+}
+
+// buildJwksURIClusters returns a list of clusters for each unique JwksUri from
+// the input list of Jwt specs. This function is to support
+// buildJwksURIClustersForProxyInstances above.
+func buildJwksURIClusters(jwtSpecs []*authn.Jwt, timeout *duration.Duration) v1.Clusters {
```

Comment:
Can you switch this to v2.Clusters ? 

We are still debating how much of v1 compat we should keep in alpha3 implementation, but for new features I would rather go directly to v2 ( or at least have first a v2 ). CDS v2 is moving ahead and early testable versions should be available in ~ 1 week.

Reply:
I don't see example of v2 cluster here yet (all the build cluster in config.go still calling v1, if I understand correctly). This file is duplicated in here so that it can change to output v2 without affect v1/deprecate version.